google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkUpdateService.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+import com.google.cloud.tools.intellij.GctFeature;
+import com.google.cloud.tools.intellij.service.PluginInfoService;
+import com.google.common.annotations.VisibleForTesting;
+import com.intellij.notification.Notification;
+import com.intellij.openapi.components.ServiceManager;
+import java.awt.event.ActionListener;
+import java.time.Clock;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/** Schedules and manages automatic updates for {@link ManagedCloudSdkService}. */
+public class ManagedCloudSdkUpdateService {
+  // one week.
+  @VisibleForTesting static final long SDK_UPDATE_INTERVAL_MS = 1000 * 60 * 60 * 24 * 7;
+  // 20 seconds to show notification and then proceed with update.
+  private static final int UPDATE_NOTIFICATION_DELAY_MS = 1000 * 20;
```

Comment:
this variable represents how long the notification remains active after it is shown right?

perhaps, UPDATE_NOTIFICATION_TTL_MS? or if you don't like the use of TTL here, then "EXPIRATION_TIME" or something like that?

Reply:
Both sounds clear enough, if you don't object, I'll use `...EXPIRATION_TIME...`.